### ISSUE TITLE
Error: "Cannot find module '@babel/core'" When Building for NODE_ENV=production...

### ISSUE THREAD

**Issue by busticated** [gatsbyjs/gatsby#19268]
## Description

I'm hitting a build error when running under `NODE_ENV=production` - specifically:

```
 ERROR #98123  WEBPACK

Generating JavaScript bundles failed

Cannot find module '@babel/core'
 babel-loader@8 requires Babel 7.x (the package '@babel/core'). If you'd like to use Babel 6.x ('babel-core'), you should install 'babel-loader@7'.

File: .cache/production-app.js
```

...but when i check my installed modules, i see:

```
npm list @babel/core babel-loader
mysite@1.0.0 /Users/me/Sites/mysite
├─┬ @svgr/webpack@4.3.3
│ ├── @babel/core@7.6.2 
│ └─┬ @svgr/plugin-jsx@4.3.3
│   └── @babel/core@7.6.2 
└─┬ gatsby@2.15.29
  ├── @babel/core@7.6.2 
  └── babel-loader@8.0.6 
```

...which seems like it'd address the `Cannot find module '@babel/core'
 babel-loader@8 requires Babel 7.x` bit... 🤷‍♂ 

### Steps to reproduce

run `NODE_ENV=production npm ci && gatsby build`

### Expected result

build succeeds

### Actual result

build fails due... missing dependency?

### Environment

```
  System:
    OS: macOS High Sierra 10.13.6
    CPU: (8) x64 Intel(R) Core(TM) i7-2600 CPU @ 3.40GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 8.16.1 - /usr/local/bin/node
    Yarn: 1.3.2 - /usr/local/bin/yarn
    npm: 5.10.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/local/bin/python
  Browsers:
    Chrome: 79.0.3945.16
    Safari: 13.0.3
  npmPackages:
    gatsby: ^2.15.29 => 2.15.29
    gatsby-cli: ^2.7.54 => 2.7.54
    gatsby-image: ^2.2.24 => 2.2.24
    gatsby-plugin-layout: ^1.1.10 => 1.1.10
    gatsby-plugin-netlify: ^2.1.17 => 2.1.17
    gatsby-plugin-react-helmet: ^3.1.10 => 3.1.10
    gatsby-plugin-sharp: ^2.2.28 => 2.2.28
    gatsby-plugin-sitemap: ^2.2.16 => 2.2.16
    gatsby-plugin-styled-components: ^3.1.8 => 3.1.8
    gatsby-plugin-svgr: ^2.0.2 => 2.0.2
    gatsby-source-contentful: ^2.1.45 => 2.1.45
    gatsby-source-filesystem: ^2.1.29 => 2.1.29
    gatsby-transformer-sharp: ^2.2.20 => 2.2.20
    gatsby-transformer-yaml: ^2.2.12 => 2.2.12
```

**Comment by LekoArts** [comment]
Hi!

Sorry to hear you're running into an issue. To help us best begin debugging the underlying cause, it is incredibly helpful if you're able to create a [minimal reproduction][reproduction]. This is a simplified example of the issue that makes it clear and obvious what the issue is and how we can begin to debug it.

If you're up for it, we'd very much appreciate if you could [provide a minimal reproduction][reproduction] and we'll be able to take another look.

Thanks for using Gatsby! 💜

[reproduction]: https://gatsby.dev/reproduction

**Comment by Marchie** [comment]
I've experienced the same issue today.

For reasons unknown, @babel/core ended up being flagged as a dev dependency in my `package-lock.json` file. Deleting and recreating the `package-lock.json` with `npm install --production` set @babel/core as a normal dependency, after which the production build worked fine. 

Once I did this, @babel/core remained as a normal dependency in `package-lock.json` after running `npm install` without the `--production` flag, too.
